### ISSUE TITLE
Add sections to break up large components [COME SUGGEST COMPONENTS!]

### ISSUE THREAD

**Issue by Frooxius** [Yellow-Dog-Man/Resonite-Issues#2968]
### Is your feature request related to a problem? Please describe.

Recently I've implemented ability for Resonite inspector to split elements on components into sections, to make visually parsing components with lots of fields easier.

This has been used for a number of new components, but there are many existing ones which could benefit from this feature.

### Describe the solution you'd like

Organize fields in following components into sections to make working with them easier:
- [ ] ParticleStyle

**We're looking for more component suggestions for this list!**

If there's components that you use often enough that they could benefit from this, feel free to suggest them below! We'll see about adding them to the list (if it's not obvious why given component would benefit from this, it might be worth adding a little blurb).

### Describe alternatives you've considered

N/A

### Additional Context

I haven't really went through the components yet and I can only think of ParticleStyle off the top of my head right now, so I just wanted to open this to help get suggestions from the community!

**Comment by ohzee00** [comment]
I'm gonna throw my hat in and suggest DynamicBoneChain!

Breaking it up by headers would help greatly in tweaking or finding certain options, heck I sometimes find a new option inbetween all of them despite messing with that component so often.

**Comment by lxw404** [comment]
`PhysicalLocomotion`, `TeleportLocomotion`, and `CharacterController`.  Then to a lesser priority: `NoClipLocomotion`, `GrabWorldLocomotion`, and `SlideLocomotion` for consistency.

**Comment by TisFoolish** [comment]
When you say "categories", what is the visual change? Is it just horizontal lines in the component that break up sections, buttons to only display certain sections, something else?

**Comment by epicEaston197** [comment]
> When you say "categories", what is the visual change? Is it just horizontal lines in the component that break up sections, buttons to only display certain sections, something else?

They look like this they've been added to newer components but not older ones yet
![image](https://github.com/user-attachments/assets/3b8bdbf5-1080-4d49-8012-8d15472b0435)

**Comment by ko-tengu** [comment]
Skinned Mesh Renderers would benefit from this quite a bit. Your average avatar head has a couple dozen blendshapes and it can be a lot to scroll through.

**Comment by Frooxius** [comment]
@ko-tengu It cannot help with that unfortunately, because those blendshapes come as part of the mesh data, not the component.

**Comment by ultrawidegamer** [comment]
VRIK and Hand Poser

**Comment by Dusty-Sprinkles** [comment]
potentially too off topic, but if this is open to component quality of life in general, a larger font size/box for the Comment component would be nice (though I could throw up a new issue if that doesn't really work here)

**Comment by Frooxius** [comment]
That'd be a separate issue.

**Comment by Redd56** [comment]
Id say character controller. 
Also sidenote i love how photon dust was implimented as a modular system, i would love to see the ridgidbody system implimented like that as well. Maybe making character controllers just a confguration of that Ridgid body system to allow for more flexability with it

**Comment by Zyro1331** [comment]
I think it would be great to see the XiexeToonMaterial broken up a little bit. Since it's a material that has _a lot_ of options by nature, and would probably make it easier to understand which parameters are related to specific features once it gains proper category support.

**Comment by Redd56** [comment]
once the shader consolidation is complete, if it does indeed make it a gigashader instead of just a bunch of small ones id love to see it implimented like photon dust as well with modules for parts that you want as needed instead of xixie/virk like
